include/seqan3/range/view/translation.hpp

```diff
@@ -0,0 +1,350 @@
+//
+//     * Redistributions of source code must retain the above copyright
+//       notice, this list of conditions and the following disclaimer.
+//     * Redistributions in binary form must reproduce the above copyright
+//       notice, this list of conditions and the following disclaimer in the
+//       documentation and/or other materials provided with the distribution.
+//     * Neither the name of Knut Reinert or the FU Berlin nor the names of
+//       its contributors may be used to endorse or promote products derived
+//       from this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL KNUT REINERT OR THE FU BERLIN BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+// LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+// OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+// DAMAGE.
+//
+// ============================================================================
+
+/*!\file
+ * \ingroup view
+ * \author Sara Hetzel <sara.hetzel AT fu-berlin.de>
+ * \brief Provides seqan3::view::translate, seqan3::view::translate_single and seqan3::view::translate_frames.
+ */
+
+#pragma once
+
+#include <vector>
+#include <stdexcept>
+
+#include <seqan3/alphabet/nucleotide/dna5.hpp>
+#include <seqan3/alphabet/aminoacid/aa27.hpp>
+#include <seqan3/alphabet/translation/translation.hpp>
+#include <seqan3/alphabet/translation/translation_details.hpp>
+#include <seqan3/range/concept.hpp>
+#include <seqan3/range/view/complement.hpp>
+
+#include <range/v3/view/chunk.hpp>
+#include <range/v3/view/concat.hpp>
+#include <range/v3/view/drop.hpp>
+#include <range/v3/view/reverse.hpp>
+#include <range/v3/view/single.hpp>
+#include <range/v3/view/take_while.hpp>
+#include <range/v3/view/transform.hpp>
+
+namespace seqan3::detail
+{
+//!\privatesection
+/*!\brief Lambda for translation of one nucleotide triplet into a single aminoacid.
+ * \details Used by view::translate_single.
+ */
+auto translate = [] (auto triplet)
+{
+    return translate_triplet(triplet[0], triplet[1], triplet[2]);
+};
+
+struct translate_single_fn
+{
+    /*!\brief Single frame translation from nucleotide to aminoacid.
+     * \tparam irng_t The type of the range being processed.
+     * \param irange The range being processed.
+     */
+    template <typename irng_t>
+        requires input_range_concept<irng_t>/* && nucleotide_concept<ranges::value_type_t<std::decay_t<irng_t>>>*/
```

Comment:
I have to correct myself again: `range_reference_t` – and no `type` inbetween!
(not sure why they made extra `range_` metafunctions, but oh well)